hdf5-sys/src/h5vl.rs

```diff
@@ -48,3 +88,1918 @@ extern "C" {
+            ) -> herr_t,
+        >,
+        pub free: Option<extern "C" fn(info: *mut c_void) -> herr_t>,
+        pub to_str: Option<extern "C" fn(info: *const c_void, str: *mut *mut c_char) -> herr_t>,
+        pub from_str: Option<extern "C" fn(str: *const c_char, info: *mut *mut c_void) -> herr_t>,
+    }
+
+    #[repr(C)]
+    #[derive(Debug)]
+    pub struct H5VL_wrap_class_t {
+        pub get_object: Option<extern "C" fn(obj: *const c_void) -> *mut c_void>,
+        pub get_wrap_ctx:
+            Option<extern "C" fn(obj: *const c_void, wrap_ctx: *mut *mut c_void) -> herr_t>,
+        pub wrap_object: Option<
+            extern "C" fn(
+                obj: *mut c_void,
+                obj_type: H5I_type_t,
+                wrap_ctx: *mut c_void,
+            ) -> *mut c_void,
+        >,
+        pub unwrap_object: Option<extern "C" fn(obj: *mut c_void) -> *mut c_void>,
+        pub free_wrap_ctx: Option<extern "C" fn(wrap_ctx: *mut c_void) -> herr_t>,
+    }
+
+    #[repr(C)]
+    #[derive(Debug, Copy, Clone, PartialEq, Eq)]
+    pub enum H5VL_attr_get_t {
+        H5VL_ATTR_GET_ACPL,
+        H5VL_ATTR_GET_INFO,
+        H5VL_ATTR_GET_NAME,
+        H5VL_ATTR_GET_SPACE,
+        H5VL_ATTR_GET_STORAGE_SIZE,
+        H5VL_ATTR_GET_TYPE,
+    }
+
+    #[repr(C)]
+    #[derive(Debug)]
+    pub struct H5VL_attr_get_args_t_union_get_acpl {
+        pub acpl_id: hid_t,
+    }
+
+    #[repr(C)]
+    #[derive(Debug)]
+    pub struct H5VL_attr_get_args_t_union_get_space {
+        pub space_id: hid_t,
+    }
+
+    #[repr(C)]
+    #[derive(Debug)]
+    pub struct H5VL_attr_get_args_t_union_get_storage_size {
+        pub data_size: *mut hsize_t,
+    }
+
+    #[repr(C)]
+    #[derive(Debug)]
+    pub struct H5VL_attr_get_args_t_union_get_type {
+        pub type_id: hid_t,
+    }
+
+    #[repr(C)]
+    #[derive(Debug)]
+    pub struct H5VL_attr_get_name_args_t {
+        pub loc_params: H5VL_loc_params_t,
+        pub buf_size: size_t,
+        pub buf: *mut c_char,
+        pub attr_name_len: *mut size_t,
+    }
+
+    #[repr(C)]
+    #[derive(Debug)]
+    pub struct H5VL_attr_get_info_args_t {
+        pub loc_params: H5VL_loc_params_t,
+        pub attr_name: *const c_char,
+        pub ainfo: *mut H5A_info_t,
+    }
+
+    #[repr(C)]
+    pub union H5VL_attr_get_args_t_union {
+        pub get_acpl: ManuallyDrop<H5VL_attr_get_args_t_union_get_acpl>,
+        pub get_info: ManuallyDrop<H5VL_attr_get_info_args_t>,
+        pub get_name: ManuallyDrop<H5VL_attr_get_name_args_t>,
+        pub get_space: ManuallyDrop<H5VL_attr_get_args_t_union_get_space>,
+        pub get_storage_size: ManuallyDrop<H5VL_attr_get_args_t_union_get_storage_size>,
+        pub get_type: ManuallyDrop<H5VL_attr_get_args_t_union_get_type>,
+    }
+
+    #[repr(C)]
+    pub struct H5VL_attr_get_args_t {
+        pub op_type: H5VL_attr_get_t,
+        pub args: H5VL_attr_get_args_t_union,
+    }
+
+    impl Debug for H5VL_attr_get_args_t {
```

Comment:
Maybe instead just `impl Debug` for the union? And then you can simply derive Debug for this struct

Reply:
`impl Debug` allows the use of the tag to decipher the union, but could could prefer a dummy impl `f.write("<UNION>")` instead

Reply:
Ahh yea, I'm being dumb, the union is untagged so we can't format it. Sorry, I'm probably just too used to Rust enums 😄 

All good then.